.github/workflows/build_test.yml

```diff
@@ -32,6 +32,21 @@ jobs:
         python-version: ${{ env.PYTHON_VERSION }}
     - uses: ./.github/actions
 
+  lint:
+    needs: smoke_test
```

Comment:
I think we can run lint irrespective of `smoke_test`?

Reply:
Well, do we want a lint failure to fail the whole build or just provide one more data point about the PR? On PyTorch they choose to end the CI early since anyway the submitter will have to fix the PR, so the whole CI will have to run again. That perspective would make me think it should run first, and everything else be conditional on it. On the otherhand, the CI might fail for other reasons so the linter is just icing on the cake, in which case it should not fail the whole CI run. I think I have a slight preference for the former: make it the reverse of this:
```
smoke_test:
  needs: lint
```

What do others think?

Reply:
To have an opinion, how about we run both and only continue if both succeed? That way, if you look at the PR you know whether there are clearly issues beyond style. And the smoke-test run doesn't take a huge amount of resources for NumPy.
The argument of rerunning CI is good though, even if it might be mildly annoying sometimes to get blocked by.

Reply:
I added lint and smoke as mandatory, making the summary look like [this](https://github.com/numpy/numpy/actions/runs/578882306).

Reply:
What would be nice is to group jobs and call it say `preflight`, and add `preflight` as `needs` instead of adding lint+smoke every time.

Reply:
I like this. I don't have an opinion on actions vs. azure, but either version seems like a good addition in any case.

Reply:
I like the idea of running it on both Azure and GH Actions, as the PR does now.

The `preflight` thing is a nice cleanup, but it's not essential and can be done separately.